sdks/go/pkg/beam/core/runtime/xlangx/namespace.go

```diff
@@ -81,36 +76,18 @@ func addWindowingStrategyID(c *pipepb.Components, idMap map[string]string, wid s
 	return idMap[wid]
 }
 
-func addEnvironmentID(c *pipepb.Components, idMap map[string]string, eid string, newID func(string) string) string {
-	if _, exists := idMap[eid]; exists {
-		return idMap[eid]
-	}
-
-	environment, exists := c.Environments[eid]
-	if !exists {
-		panic(errors.Errorf("attempted to add namespace to missing windowing strategy id: %v not in %v", eid, c.Environments))
-	}
-
-	idMap[eid] = newID(eid)
-
-	// Updating Environments map
-	c.Environments[idMap[eid]] = environment
-	delete(c.Environments, eid)
-
-	return idMap[eid]
-}
-
 func addNamespace(t *pipepb.PTransform, c *pipepb.Components, namespace string) {
 	newID := func(id string) string {
 		return fmt.Sprintf("%v@%v", id, namespace)
 	}
 
 	idMap := make(map[string]string)
 
-	// Update Environment ID of PTransform
-	if t.EnvironmentId != "" {
-		t.EnvironmentId = addEnvironmentID(c, idMap, t.EnvironmentId, newID)
-	}
+	// TODO: Currently environments are not namespaced. This works under the
```

Comment:
Consider including the JIRA tag for history context, or instead of TODO (which implies work that should likely be done) use Note: or avoid a prefix entirely, since all comments are notes to our future selves.

As for the content, it seems probable that we'd make sure the Go ExpansionService that handles these calls would do the namespacing on the response itself. That way it can build the pipeline graph normally, and we simply run the replacement there, where we are already certain that it's a foreign component, rather than the primary pipeline.

Reply:
Changed it to a "note". That makes more sense than making a JIRA for the vague possibility of multiple Go environments.